x-pack/plugins/data_enhanced/server/search/session/get_search_status.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ApiResponse } from '@elastic/elasticsearch';
+import { ElasticsearchClient } from 'src/core/server';
+import { isAsyncSearchStatusResponse, SearchStatus } from './types';
+
+export async function getSearchStatus(client: ElasticsearchClient, asyncId: string) {
+  try {
+    const path = encodeURI(`/_async_search/status/${asyncId}`);
+    const response: ApiResponse<any> = await client.transport.request({
+      path,
+      method: 'GET',
+    });
+    if (isAsyncSearchStatusResponse(response)) {
+      if ((response.is_partial && !response.is_running) || response.completion_status > 200) {
+        return SearchStatus.ERROR;
```

Comment:
When there is an error returned, all we know is that there was an error but we aren't preserving information that explains the error.

Maybe in a later PR, unless you feel comfortable doing it here, we should also return an error message that can be saved in a new field in BackgroundSessionSavedObjectAttributes